website/content/api-docs/secret/pki.mdx

```diff
@@ -3249,6 +3446,104 @@ The following privileged endpoints allow the operator to control information
 about the core contents of certificates and to perform privileged operations
 like rotating the CRLs or performing tidy operations.
 
+### List CEL roles
+
+Refer to the [earlier section](#list-cel-roles) for more information about
+listing CEL roles.
+
+### Create/Update CEL role {#create-update-cel-role}
+
+This endpoint creates or updates the CEL role definition. 
+
+| Method  | Path               |
+| :------ | :----------------- |
+| `POST`  | `/pki/cel/roles/:name` |
+| `PATCH` | `/pki/cel/roles/:name` |
+
+#### Parameters
+
+- `name` `(string: <required>)` - Specifies the name of the CEL role to create. This
+  is part of the request URL.
+- `cel_program`
+
+#### Sample payload
+
+```json
+{
+  "cel_program": {
+    "variables": [
+      {
+        "name": "validate_cn",
+        "expression": "has(request.common_name) && request.common_name == \"example.com\""
+      },
+      {
+        "name": "small_ttl",
+        "expression": "has(request.ttl) && duration(request.ttl) < duration(\"4h\")"
+      },
+      {
+        "name": "cn_value",
+        "expression": "request.common_name"
+      },
+      {
+        "name": "not_after",
+        "expression": "now + duration(request.ttl)"
+      },
+      {
+        "name": "cert",
+        "expression": "CertTemplate{ Subject: PKIX.Name{ CommonName: cn_value, Country: [\"ZW\", \"US\"] }, NotBefore: now, NotAfter: not_after, IsCA: true, MaxPathLen: 10, PolicyIdentifiers: [ ObjectIdentifier{ arc: [1u,2u,3u] }, ObjectIdentifier{ arc: [2u,59u,1u] } ], IPAddresses: [ net.IP{ IP: b\"\\x0A\\x00\\x00\\x00\" } ] }"
+      },
+      {
+        "name": "output",
+        "expression": "ValidationOutput{ template: cert, generate_lease: small_ttl, no_store: !small_ttl, issuer_ref: \"default\", key_type: request.key_type, key_bits: uint(request.key_bits) }"
+      },
+      {
+        "name": "err",
+        "expression": "'Request should have Common name: ' + cn_value"
```

Comment:
```suggestion
        "expression": "'Request has invalid common name'"
```

(`cn_value` is the user's input, which doesn't really tell them the expected value like the rest of the existing message seems to lead to. However, telling the actual expected value isn't a good idea either.)